### PR TITLE
chore: Update flow

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]

File: packages/medusa/src/api/auth/utils/validate-token.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const validateToken = () => {
     const req_ = req as AuthenticatedMedusaRequest
 
     if (!token) {
-      return next()
+      return res.status(401).json({ message: "Invalid token" })
```
Comment (sradevski) [inline packages/medusa/src/api/auth/utils/validate-token.ts]: I am just not sure if this stops the execution flow or not, cause IIRC you need to do `next(new Error(''))` to return with an error from a middleware.

Worst case we try to do res.status or res.send lower down the call stack and that fails

**Review by sradevski — COMMENTED** [review]
LGTM!

File: packages/medusa/src/api/auth/utils/validate-token.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/medusa/src/api/auth/utils/validate-token.ts]: @sradevski, I believe both works, as validated by the tests, but I can change to use the `new Error` to be consistent with other middleware 👍

Comment (adrien2p) [reply packages/medusa/src/api/auth/utils/validate-token.ts]: I would also go with next(new MedusaError) 👌

**Review by adrien2p — APPROVED** [review]

File: integration-tests/http/__tests__/auth/admin/auth.spec.ts [hunk 2]
```diff
@@ -264,6 +264,28 @@ medusaIntegrationTestRunner({
...
+        jest.useFakeTimers()
+
+        // Register user
+        await api.post("/auth/user/emailpass/register", {
+          email: "test@medusa-commerce.com",
+          password: "secret_password",
+        })
+
+        // Advance time by 15 minutes
+        jest.advanceTimersByTime(15 * 60 * 1000)
```
Comment (adrien2p) [inline integration-tests/http/__tests__/auth/admin/auth.spec.ts:278]: Q: do we have useRealTimers call in the before each? Just be sure that we reset everytime

**Review by adrien2p — APPROVED** [review]

**Pull request merged**